### PR TITLE
esp32-core: Add support for the external SPIFLASH

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
 - Fix some trivial, mostly cosmetic, issues in the driver.
 - Add support for the board.
## Impact
Shouldn't affect any other existing code.
## Testing
The provided defconfig was used to test.

### CODE REVIEW

File: boards/xtensa/esp32/esp32-core/Kconfig [hunk 1]
```diff
@@ -44,4 +44,23 @@ config ESP32CORE_FLASH_IMAGE
 	---help---
 		Create flash_image.bin mainly used for QEMU.
 
+choice
```
Comment (acassis) [inline boards/xtensa/esp32/esp32-core/Kconfig:47]: The SPIFLASH should work independently from the File System the user chose, maybe it is a limitation of current implementation, correct?

**Review by acassis — APPROVED** [review]

File: boards/xtensa/esp32/esp32-core/Kconfig (hunk 1 above)
Comment (Ouss4) [reply boards/xtensa/esp32/esp32-core/Kconfig:47]: We could add any other file system, I just picked some of what we have in the repo.
The board logic that was added with this PR prepares the driver to be mounted on a file system, but the SPIFLASH driver itself is completely independent from that.

**Comment by xiaoxiang781216** [comment]
@Ouss4 PR https://github.com/apache/incubator-nuttx/pull/1755 should fix the break, please rebase your change again. Since PR https://github.com/apache/incubator-nuttx/pull/1611 send out three weeks ago, It's hard to catch the new project config file added recently. Sorry for inconvenience.

**Comment by Ouss4** [comment]
@xiaoxiang781216 It looks like it passed.  The only one failing is a macOS build which is not related to that PR.  I did a similar change as #1755 when I saw your PR, I think if I didn't the build would've failed when #1611 was merged.

**Comment by v01d** [comment]
The mac fail is:

<pre>
====================================================================================
Configuration/Tool: esp32-core/random
------------------------------------------------------------------------------------
  Cleaning...
HEAD detached at pull/1754/merge
Changes not staged for commit:
	modified:   boards/xtensa/esp32/esp32-core/configs/mmcsdspi/defconfig

no changes added to commit
  Configuring...
  Building NuttX...
  Normalize esp32-core/random
HEAD detached at pull/1754/merge
Changes not staged for commit:
	modified:   boards/xtensa/esp32/esp32-core/configs/mmcsdspi/defconfig
</pre>
I think this needs a configuration refresh.

**Comment by Ouss4** [comment]
> I think this needs a configuration refresh.

That's what I did earlier.  I wonder why only the macOS one caught that, the rest didn't complain.
Anyway, I rebased, let's see again.

File: arch/xtensa/src/esp32/esp32_spiflash.c [hunk 2]
```diff
@@ -131,7 +155,7 @@ static struct esp32_spiflash_s s_esp32_spiflash1 =
             .bwrite = esp32_bwrite,
             .read   = esp32_read,
             .ioctl  = esp32_ioctl,
-#if defined(CONFIG_MTD_BYTE_WRITE)
+#ifdef CONFIG_MTD_BYTE_WRITE
```
Comment (davids5) [inline arch/xtensa/src/esp32/esp32_spiflash.c:158]: @Ouss4 - Curiosity question: Why would the the `#ifdef` be preferred over the `#if defined`  ?

Comment (Ouss4) [reply arch/xtensa/src/esp32/esp32_spiflash.c:158]: @davids5 That was originally `#if defined(CONFIG_MTD_BYTE_WRITE) && defined(ANOTHER_CFG)` that other config was just a copy/paste typo when porting, that doesn't apply to this file.
I find "#ifdef" a little explicit compared to `#defined(ONLY_ONE_CFG)` but the two are equivalent in this case.

**Pull request merged**